include/systems/condensed_eigen_system.h

```diff
@@ -152,6 +152,12 @@ class CondensedEigenSystem : public EigenSystem
    */
   void copy_super_to_sub(const SparseMatrix<Number> & super, SparseMatrix<Number> & sub);
 
+  /**
+   * Instructs not to create the condensed submatrices from the global matrices in the solve
```

Comment:
Do we definitely understand *why* we might need this?  Is it's just because it forces us to `close()` earlier and thereby trigger PETSc's "you gave us a sparsity pattern but since we see zeroes we're going to assume you gave us the wrong sparsity pattern and shrink it" policy?

If so, some documentation of that here would be good.

Also if so, is there still no better workaround at this point?  Feels like someone should have eventually added a `PetscDidIStutter(false)` we could be calling...

Reply:
The reason to not create submatrices in the solve is that the submatrix creation will create an entirely new matrix and consequently wipe away the callbacks I had attached to the condensed matrix. SLEPc will not be able to communicate with us if that is the case.

Also it seems that all previous users of this class (the RB code) have already assembled their system before handing off to the solver so `create_submatrix` is actually meaningful for those users. We (MOOSE) generally do not do assembly until we are within the solve so `create_submatrix` is truly useless. It's creating a submatrix from a matrix with no fill